### DISCUSSION TITLE
Using useAsyncData and Pinia

### DISCUSSION THREAD

**Discussion by pedrodruviaro** [nuxt/nuxt#24909] [Questions] [answered]
I'm new to Nuxt and I'm building an app with Supabase and Pinia. In my store, I have the following function:

```
async function getTransactions() {
    try {
        pending.value = true;

        const { data, error } = await supabase
            .from("transactions")
            .select()
            .eq("user_id", user.value.id)
            .order("date", { ascending: false });

        if (error) {
            throw new Error();
        }

        transactions.value = data;

        return { error: null };
    } catch (error) {
        notifyError();
    } finally {
        pending.value = false;
    }
}
```
In my component, I use useAsyncData like this:

```
await useAsyncData("transactions", () => {
    transactionsStore.getTransactions();
});
```

Is this the best way? Since the "getTransactions" function is also called after a creation form, this was the only way that didn't result in errors in the console. I tried using useAsyncData within the store, but it caused even more problems for me. I still hesitate on how to proceed.

**Comment by TheAlexLichter** [comment, marked as answer]
The approach is fine IMO and what is suggested in the [Pinia docs](https://pinia.vuejs.org/ssr/nuxt.html#Awaiting-for-actions-in-pages). The only thing you want to do is *return* something inside the async function of `useAsyncData`.

You **don't** want to use `useAsyncData` in the store due various factors, one of them being that pinia hydrates on it's own and useAsyncData/useFetch does too. Instead, you might want to check [callOnce](https://github.com/nuxt/nuxt/pull/24787) if you don't need reactivity!
